lib/materials/data_helpers.rb

```diff
@@ -46,6 +46,8 @@ def materials_data( materials,
         has_activities = material.respond_to?(:activities) && !material.activities.nil?
         has_pretest = material.respond_to?(:has_pretest) && material.has_pretest
 
+        saves_student_data = material.respond_to?(:saves_student_data) && material.saves_student_data
+
```

Comment:
If the material doesn't respond to :saves_student_data, then it should default to `true`. Otherwise on old databases it will look like all of the materials do not save student data.